packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -86,6 +88,15 @@ class AnalyzeCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    await runCommandAndStreamOutput(
```

Comment:
This will definitely work, but it will mean a lot of redundant invocations of `flutter update-packages`.

Also, this won't be productive if the user is not using the flag `--flutter-repo`.

Since it looks like any time you analyze a flutter project, we're running pub, maybe running `flutter update-packages` every time we `flutter analyze --flutter-repo` would be ok?

Reply:
I agree with you since running `pub get` is part of the flow, it would probably be okay to run `flutter update-packages` when running `flutter analyze --flutter-repo`.
I'll add the check for it.